app/rtc_compare.py

```diff
@@ -0,0 +1,488 @@
+
+def _get_parser():
+    parser = argparse.ArgumentParser(
+        description='Compare two RTC products',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    # Inputs
+    parser.add_argument('input_file',
+                        type=str,
+                        nargs=2,
+                        help='Input RTC products in NETCDF/HDF5 format')
+
+    return parser
+
+
+def _unpack_array(val_in, hdf5_obj_in):
+    '''
+    Unpack the array of array into ordinary numpy array.
+    Convert an HDF5 object reference into the path it is pointing to.
+
+    For internal use in this script.
+
+    Parameter:
+    -----------
+    val_in: np.ndarray
+        numpy array to unpack
+    hdf5_obj_in:
+        Source HDF5 object of `val_in`
+
+    Return:
+    val_out: np.ndarray
+        unpacked array
+
+    '''
+    list_val_in = list(itertools.chain.from_iterable(val_in))
+
+    list_val_out = [None] * len(list_val_in)
+    for i_val, element_in in enumerate(list_val_in):
+        if isinstance(element_in, h5py.h5r.Reference):
+            list_val_out[i_val] = np.str_(hdf5_obj_in[element_in].name)
+        else:
+            list_val_out[i_val] = element_in
+    val_out = np.array(list_val_out)
+
+    return val_out
+
+
+def print_data_difference(val_1, val_2, indent=4):
+    '''
+    Print out the difference of the data whose dimension is >= 1
+
+    Parameters:
+    -----------
+    val_1, val_2: np.array
+        Data that has difference to each other
+    indent: int
+        Number of spaces for indentation
+    '''
+
+    str_indent = ' ' * indent + '-'
+
+    # printout the difference
+    if not issubclass(val_1.dtype.type, np.number):
+        # Routine for non-numeric array
+        # Print out the first discrepancy in the array
+        flag_discrepancy = (val_1 != val_2)
+        index_first_discrepancy = np.where(flag_discrepancy)[0][0]
+        print(f'{str_indent} The first discrepancy has detected from index '
+              f'[{index_first_discrepancy}] : '
+              f'1st=({val_1[index_first_discrepancy]}), '
+              f'2nd=({val_2[index_first_discrepancy]})')
+        return None
+
+    # Routine for numeric array
+    difference_val = val_1 - val_2
+    index_max_diff = np.nanargmax(np.abs(difference_val))
+    print(f'{str_indent} Maximum difference detected from index '
+          f'[{index_max_diff}]: '
+          f'1st: ({val_1[index_max_diff]}), 2nd: ({val_2[index_max_diff]}) = '
+          f'diff: ({difference_val[index_max_diff]})')
+
+    if not (issubclass(val_1.dtype.type, np.floating) or
+            issubclass(val_1.dtype.type, np.complexfloating)):
+        return None
+
+    # Check pixel-by-pixel nan / non-nan difference
+    mask_nan_val_1 = np.isnan(val_1)
+    mask_nan_val_2 = np.isnan(val_2)
+
+    mask_nan_discrepancy = np.logical_xor(mask_nan_val_1, mask_nan_val_2)
+
+    if not np.any(mask_nan_discrepancy):
+        print(f'{str_indent} NaN discrepancy was not detected.')
```

Comment:
Do we need this line?

Reply:
I wanted to be explicit about that, so that we don't need to worry about the NaNs in when further investigating the difference. I can remove it if you want.

Reply:
Yes, thanks for adding. I think it's better we remove it to avoid overloading the user with messages.

Reply:
Did you remove "the early return" here ([94c1d37](https://github.com/opera-adt/RTC/pull/9/commits/94c1d37f6b262cd69cdc853bd874f666dc49b0e3))? I meant that only the line with `print()` that was not necessary.  Or did you do it because of another reason?

Reply:
Okay. On top of your comment (readability), I've read through few more articles, and now I think I have better understanding of "early return." I've made the change to re-engage the pattern.

Reply:
Awesome. Thank you, @seongsujeong